src/commands/inhouse.js

```diff
@@ -0,0 +1,266 @@
+                        var jsonResp = JSON.parse(data);
+                        _summonerId = jsonResp.id;
+                        resolve(_summonerId);
+                    } catch(e) {
+                        reject(e.message);
+                    }
+                    });
+            }).on("error", (err) => {
+                reject(`Got error: ${err.message}`);
+            });
+        });
+    }
+    
+    getRankWeight(summId) {
+        return new Promise((resolve, reject) => {
+            var _rankWeight = 0;
+            var tier;
+            var division;
+    
+            let fullUrl = urlSummRank + summId + urlFinish;
+            https.get(fullUrl, (resp) => {
+                let data = '';
+                resp.on('data', (chunk) => {
+                    data += chunk;
+                  });
+                resp.on('end', () => {
+                    try {
+                        var jsonResp = JSON.parse(data)
+                        var i;
+                        for(i = 0; i < 3; i++) {
+                            if(jsonResp[i] != null) {
+                                if(jsonResp[i].queueType === 'RANKED_SOLO_5x5') {
+                                    tier = jsonResp[i].tier;
+                                    division = jsonResp[i].rank;
+                                    break;
+                                }
+                            }
+                        }
+                        _rankWeight = utils.calculateSummonerRankWeight(tier, division);
+                        resolve(_rankWeight);
+                    } catch(e) {
+                        reject(e.message);
+                    }
+                  });
+            }).on("error", (err) => {
+                reject(`Got error: ${err.message}`);
+            });
+        });
+       
+    }
+}
+
+class Match {
+    constructor(players, _type) {
+        if(players.length != 10) {
+            console.error("need 10 players");
+            return;
+        }
+
+        this.team1 = [];
+        this.team2 = [];
+        this.type = _type;
+
+        this.sortTeams(players);
+    }
+
+    sortTeams(players) {
+        if(this.type === '4fun') {
+            players.forEach(element => {
+                let teamAssignment = Math.random(1, 3);
+                if(teamAssignment === 1) {
+                    if(this.team1.length < 5) {
+                        this.team1.push(element);
+                    } else {
+                        this.team2.push(element);
+                    }
+                } else {
+                    if(this.team2.length < 5) {
+                        this.team2.push(element);
+                    } else {
+                        this.team1.push(element);
+                    }
+                }
+            });
+        } else {
+            players.sort(function(a, b) {
+                return b.rankWeight - a.rankWeight;
+            });
+
+            for(var i = 0; i < players.length; i++) {
```

Comment:
This chunk of code seems a little verbose when you write it out imperatively. 
Here is a possible replacement. This is a bit nitpicky, let me know your thoughts. 
```javascript
// player1 = [1,2,3,4,5,6,7,8,9)
  team1 = players.filter(e => e % 2)
  team2 = players.filter(e => !(e % 2))

  console.log(team1) // [ 1, 3, 5, 7, 9 ]
  console.log(team2) // [ 2, 4, 6, 8 ]
```